examples/webgl_meshline.html

```diff
@@ -0,0 +1,221 @@
+
+				}
+
+				geometry.addAttribute( 'position', new THREE.Float32BufferAttribute( vertices, 3 ) );
+				geometry.addAttribute( 'color', new THREE.Float32BufferAttribute( colors, 3 ) );
+
+				// materials
+
+				material1 = new THREE.MeshBasicMaterial( { color: 0xffffff, meshLine: true, meshLineWidth: 10 } );
+				material2 = new THREE.MeshBasicMaterial( { vertexColors: THREE.VertexColors, meshLine: true, meshLineWidth: 5 } );
+				material3 = new THREE.MeshBasicMaterial( { color: 0xffffff, meshLine: true, meshLineWidth: 10 } );
+
+				// mesh lines
+
+				var meshLine = new THREE.MeshLine( geometry, material1 );
+				meshLine.position.set( - 200, 0, 0 );
+				scene.add( meshLine );
+
+				meshLine = new THREE.MeshLine( geometry, material2 );
+				scene.add( meshLine );
+
+				meshLine = new THREE.MeshLine( geometry, material3, widthCallback );
+				meshLine.position.set( 200, 0, 0 );
+				scene.add( meshLine );
+
+				// event listener
+
+				document.addEventListener( 'mousemove', onDocumentMouseMove, false );
+				document.addEventListener( 'touchstart', onDocumentTouchStart, false );
+				document.addEventListener( 'touchmove', onDocumentTouchMove, false );
+
+				window.addEventListener( 'resize', onWindowResize, false );
+
+			}
+
+			function widthCallback( p ) {
+
+				return 2 + 1.5 * Math.sin( 50 * p ); // this will create a wavy curve effect
+
+			}
+
+			function onWindowResize() {
+
+				windowHalfX = window.innerWidth / 2;
+				windowHalfY = window.innerHeight / 2;
+
+				camera.aspect = window.innerWidth / window.innerHeight;
+				camera.updateProjectionMatrix();
+
+				renderer.setSize( window.innerWidth, window.innerHeight );
+
+			}
+
+			//
+
+			function onDocumentMouseMove( event ) {
+
+				mouseX = event.clientX - windowHalfX;
+				mouseY = event.clientY - windowHalfY;
+
+			}
+
+			function onDocumentTouchStart( event ) {
+
+				if ( event.touches.length > 1 ) {
+
+					event.preventDefault();
+
+					mouseX = event.touches[ 0 ].pageX - windowHalfX;
+					mouseY = event.touches[ 0 ].pageY - windowHalfY;
+
+				}
+
+			}
+
+			function onDocumentTouchMove( event ) {
+
+				if ( event.touches.length == 1 ) {
+
+					event.preventDefault();
+
+					mouseX = event.touches[ 0 ].pageX - windowHalfX;
+					mouseY = event.touches[ 0 ].pageY - windowHalfY;
+
+				}
+
+			}
+
+			function animate() {
+
+				requestAnimationFrame( animate );
+				render();
+
+			}
+
+			function render() {
+
+				camera.position.x += ( mouseX - camera.position.x ) * .05;
+				camera.position.y += ( - mouseY + 200 - camera.position.y ) * .05;
+
+				camera.lookAt( scene.position );
+
+				var time = Date.now() * 0.0005;
+
+				for ( var i = 0; i < scene.children.length; i ++ ) {
+
+					var object = scene.children[ i ];
+
+					if ( object instanceof THREE.Line ) {
+
+						object.rotation.y = time * ( i % 2 ? 1 : -1 );
+
+					}
+
+				}
+
+				// animate line width
+
+				material1.meshLineWidth = 10 * ( 1 + 0.5 * Math.sin( 5 * time ) );
```

Comment:
Perhaps also set size attenuation on the first one.

What are the units of `width`?